runtime/jvmti/jvmtiClass.c

```diff
@@ -366,13 +366,29 @@ jvmtiGetClassSignature(jvmtiEnv* env,
 					signature[arity] = 'L';
 					memcpy(signature + arity + 1, J9UTF8_DATA(utf), utfLength);
 					signature[allocSize - 1] = ';';
+
+					if (J9ROMCLASS_IS_ANON_OR_HIDDEN(leafType->romClass)) {
+						int i = allocSize - 3;
```

Comment:
use IDATA instead of int

Reply:
if ';' is at `- 1` you can probably start from `- 4`

Reply:
actually, do we need to scan at all? doesnt `ROM_ADDRESS_LENGTH` indicate where the rom address starts?

Reply:
we should make a helper for this

Reply:
Good point, I'll use ROM_ADDRESS_LENGTH instead of scanning

Reply:
updated in 3a22916